### ISSUE TITLE
Upgrading to setup-terraform 2.x breaks previously working `terraform validate` invocation

### ISSUE THREAD

**Issue by oscep** [hashicorp/setup-terraform#276]
Our GitHub Action performing `terraform validate` fails with an obscure error message on a commit it previously validated successfully:

    Run terraform validate -no-color
    /home/runner/work/_temp/12cb240b-5a9f-44d7-9df4-0c2e0c97750d/terraform-bin validate -no-color
    /home/runner/work/_temp/12cb240b-5a9f-44d7-9df4-0c2e0c97750d/terraform:4211
      core.setOutput('exitcode', exitCode.toString(10));
                                          ^
    
    TypeError: Cannot read properties of null (reading 'toString')
        at /home/runner/work/_temp/12cb240b-5a9f-44d7-9df4-0c2e0c97750d/terraform:4211:39
    Error: Process completed with exit code 1.

Here is our `.github/workflows/terraform_pr.yml`:

    name: Terraform PR Check
    
    on: [pull_request]
    
    jobs:
      terraform:
        runs-on: ubuntu-latest
        steps:
          - name: Checkout
            uses: actions/checkout@v2
    
          - name: HashiCorp - Setup Terraform
            uses: hashicorp/setup-terraform@v2
            with:
              terraform_version: 1.3.4
    
          - name: Terraform fmt
            id: fmt
            run: terraform fmt -check -recursive
    
          - name: Terraform Init
            id: init
            run: terraform init -backend=false
    
          - name: Terraform Validate
            id: validate
            run: terraform validate -no-color

Terraform can successfully validate our code if I downgrade `hashicorp/setup-terraform` to 1.3.2:
    
    Run hashicorp/setup-terraform@v1.3.2
    /usr/bin/unzip -q /home/runner/work/_temp/a8c797cf-3a46-47e8-a351-19b399db45c4    

    <SNIP>

    Run terraform validate -no-color
    /home/runner/work/_temp/372541a7-9c9f-4a67-9c3d-5ba558003d1f/terraform-bin validate -no-color
    
    Warning: Argument is deprecated
    
      with module.shared.aws_s3_bucket.terraform-bucket,
      on system/account_project.tf line 31, in resource "aws_s3_bucket" "terraform-bucket":
      31: resource "aws_s3_bucket" "terraform-bucket" {
    
    Use the aws_s3_bucket_versioning resource instead
    
    (and 128 more similar warnings elsewhere)
    Success! The configuration is valid, but there were some validation warnings
    as shown above.
    
    Warning: The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
    Warning: The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
    Warning: The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

Could this issue be caused by terraform-setup still using `set-output`?

**Comment by oscep** [comment]
Running the check with debug output enabled leads to this message:

    ##[debug]Terraform exited with code null.

![Screenshot 2022-12-02 at 15 18 31](https://user-images.githubusercontent.com/117340990/205314257-3480e493-8aaa-4ecb-8d8c-c92e35da60ad.png)

**Comment by kishaningithub** [comment]
i am also getting the sam error any workarounds or fixes?

**Comment by kishaningithub** [comment]
By looking at the code i see that the `exitCode` variable is populated by the result of `exec` node js function call

https://github.com/hashicorp/setup-terraform/blob/1fdd4cd311e198fb45135f3a68294d3df2d5bda8/wrapper/terraform.js#L33

These are the reasons due to which node js `exec` call can return a null result
- https://stackoverflow.com/a/74086872/3316017
- https://sathishsuresh.medium.com/node-js-child-process-exec-exit-code-null-d29838b9f309

**Comment by bflad** [comment]
@kishaningithub do you believe your Terraform CLI command would've generated more than 1MB of output? Regardless, it seems reasonable to increase the command execution `maxBuffer` configuration in this case to try and prevent confusion if it would help reduce the possibility for this type of NodeJS error. I think we'd be happy to review that sort of contribution.

One potential workaround in this case is when the GitHub Actions outputs are not needed from the command would be to disable the NodeJS wrapper `terraform_wrapper: false` in the workflow configuration -- the command output is still available as text in the GitHub Actions logs and a non-zero command exit status would still fail the workflow.

**Comment by danielkubat** [comment]
I'm facing same issue as described above (node 20, setup-terraform v3)

![image](https://github.com/hashicorp/setup-terraform/assets/42444343/eca52aba-fac7-4c4d-82b2-79b16d20fd65)

**Comment by kishaningithub** [comment]
@danielkubat Is the above on a public github runner?

**Comment by danielkubat** [comment]
@kishaningithub nope, self-hosted runner on GKE (using ARC) with 8GB request/limit per pod.

**Comment by kishaningithub** [comment]
@danielkubat Can you check the logs (kernlog, dmesg etc) and find what killed the terraform process ?

If you think it was an OOM killer can you try beeffing up the specs atleast temporarily?

**Comment by danielkubat** [comment]
> @kishaningithub nope, self-hosted runner on GKE (using ARC) with 8GB request/limit per pod.

...but, I have multiple groups, and didn't set runner group correctly in action. In other words, **issue was related to lack of memory**. @kishaningithub you was right, thanks for your help.
